examples/rhat_monitor.cpp

```diff
@@ -270,26 +235,24 @@ void debug_print(double variance_of_means, double mean_of_variances,
   std::cout << std::endl;
 }
 
-static void controller_loop(std::vector<Queue>& queues,
+static void controller_loop(std::vector<AtomicChainStats>& chain_statses,
                             std::vector<std::jthread>& workers,
                             double rhat_threshold, std::latch& start_gate,
                             std::size_t max_draws_per_chain,
-			    std::stop_source& stopper) {
+                            std::stop_source& stopper) {
   interactive_qos();
   start_gate.wait();
-  const std::size_t M = queues.size();
+  const std::size_t M = chain_statses.size();
   std::vector<double> chain_means(M, std::numeric_limits<double>::quiet_NaN());
   std::vector<double> chain_variances(M,
                                       std::numeric_limits<double>::quiet_NaN());
   std::vector<std::size_t> counts(M, 0);
   while (true) {
     for (std::size_t m = 0; m < M; ++m) {
-      ChainStats u;
-      while (queues[m].pop(u)) {
-        chain_means[m] = u.sample_mean;
-        chain_variances[m] = u.sample_var;
-        counts[m] = u.count;
-      }
+      ChainStats u = chain_statses[m].load();
```

Comment:
Without some flag here to tell the main thread it has taken the workers info already you could potentially be pulling in old info that you have already taken